File: product-processor-service/src/main/java/com/example/product_processor_service/service/common/kafka/template/router/KafkaTemplateRouter.java
package com.example.product_processor_service.service.common.kafka.template.router;

import org.springframework.kafka.core.KafkaTemplate;

import java.io.Serializable;

public interface KafkaTemplateRouter {
    <V extends Serializable> KafkaTemplate<String, V> route(Class<V> clazz);
}

File: product-manager-service/src/main/kotlin/com/example/product_manager_service/service/kafka/template/router/KafkaTemplateRouter.kt
package com.example.product_manager_service.service.kafka.template.router

import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Service
import java.io.Serializable
import kotlin.reflect.KClass

@Service
interface KafkaTemplateRouter {
    fun <T : Serializable> route(valueClass: KClass<T>): KafkaTemplate<String, T>
}